corehq/apps/dump_reload/interface.py

```diff
@@ -46,24 +47,33 @@ def slug(self):
         raise NotImplementedError
 
     @abstractmethod
-    def load_objects(self, object_strings, force=False):
+    def load_objects(self, object_strings, force=False, dry_run=False):
         """
         :param object_strings: iterable of JSON encoded object strings
         :param force: True if objects should be loaded into an existing domain
         :return: loaded object Counter
         """
         raise NotImplementedError
 
-    def load_from_file(self, extracted_dump_path, dump_meta, force=False):
-        file_path = os.path.join(extracted_dump_path, '{}.gz'.format(self.slug))
+    def load_from_path(self, extracted_dump_path, dump_meta, force=False, dry_run=False):
```

Comment:
This allows the loader to load any file as long as it matches the filename pattern `{slug}*.gz`. This means that the dump can be split into multiple pieces instead of having to be in a single gzip file.

Reply:
I guess we have few enough dumpers now that this isn't an issue, but it might be prudent to guard against false positives here, such as if one dumper has a slug that is contained in the start of another slug, like `sql` and `sql-products` or something.

Reply:
good point - I can add a quick test case to prevent that from happening.

Reply:
aa5d472